samples/csharp/end-to-end-apps/ObjectDetection-Onnx/README.md

```diff
@@ -0,0 +1,194 @@
+
+### The solution contains three projects
+
+- [**OnnxObjectDetection**](./OnnxObjectDetection) is a .NET Standard library used by both the WPF app and the Web app.  It contains most of the logic for running images trough the model and parsing the resulting prediction.  This project also contains the Onnx model file.  With the exception of drawing the labels bounding boxes on the image/screen, all of the following code snippets are contained in this project.
+- [**OnnxObjectDetectionWeb**](./OnnxObjectDetectionWeb) contains an ASP.NET Core Web App that that contains both **Razor UI pages** and an **API controller** to process and render images.
+- [**OnnxObjectDetectionApp**](./OnnxObjectDetectionApp) contains an .NET CORE WPF Desktop App that uses [OpenCvSharp](https://github.com/shimat/opencvsharp) to capture the video from the devices webcam.
+
+## Code Walkthrough
+
+_This sample differs from the [getting-started object detection sample](https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/getting-started/DeepLearning_ObjectDetection_Onnx) in that here we load/process the images **in-memory** whereas the getting-started sample loads the images from a **file**._
+
+Create a class that defines the data schema to use while loading data into an `IDataView`. ML.NET supports the `Bitmap` type for images, so we'll specify `Bitmap` property decorated with the `ImageTypeAttribute`, as shown below.
+
+```csharp
+public class ImageInputData
+{
+    [ImageType(416, 416)]
+    public Bitmap Image { get; set; }
+}
+```
+
+### ML.NET: Configure the model
+
+The first step is to create an empty DataView as we just need schema of data while configuring up model.
+
+```csharp
+var dataView = _mlContext.Data.LoadFromEnumerable(new List<ImageInputData>());
+```
+
+The second step is to define the estimator pipeline. Usually when dealing with deep neural networks, you must adapt the images to the format expected by the network. For this reason, the code below resizes and transforms the images (pixel values are normalized across all R,G,B channels).
+
+```csharp
+var pipeline = _mlContext.Transforms.ResizeImages(resizing: ImageResizingEstimator.ResizingKind.Fill, outputColumnName: "image", imageWidth: ImageSettings.imageWidth, imageHeight: ImageSettings.imageHeight, inputColumnName: nameof(ImageInputData.Image))
+                .Append(_mlContext.Transforms.ExtractPixels(outputColumnName: "image"))
+                .Append(_mlContext.Transforms.ApplyOnnxModel(modelFile: onnxModelFilePath, outputColumnNames: new[] { TinyYoloModelSettings.ModelOutput }, inputColumnNames: new[] { TinyYoloModelSettings.ModelInput }));
+```
+
+You also need to inspect the neural network to get the names of the input and output nodes. To do this, you can use tools like [Netron](https://github.com/lutzroeder/netron), which is automatically installed with [Visual Studio Tools for AI](https://visualstudio.microsoft.com/downloads/ai-tools-vs/).
```

Comment:
Is Model Builder (which is where this link leads) different than Visual Studio Tools for AI? If we're just referring to Model Builder, maintain consistency and just call it Model Builder.

```suggestion
You also need to inspect the neural network to get the names of the input and output nodes. To do this, you can use tools like [Netron](https://github.com/lutzroeder/netron), which is automatically installed with [Model Builder](https://visualstudio.microsoft.com/downloads/ai-tools-vs/), a VS extension to make ML more approachable.
```